feature_editorial/src/main/java/cm/aptoide/pt/feature_editorial/data/AptoideEditorialRepository.kt

```diff
@@ -20,14 +20,17 @@ internal class AptoideEditorialRepository @Inject constructor(
   private val storeName: String
 ) : EditorialRepository {
 
+  private val cachedUrls = mutableMapOf<String, String>()
+
   override suspend fun getLatestArticle(): List<ArticleMeta> =
     editorialRemoteDataSource.getLatestEditorial()
       .datalist?.list?.map(EditorialJson::toDomainModel) ?: throw IllegalStateException()
 
-  override suspend fun getArticle(widgetUrl: String): Article =
-    editorialRemoteDataSource.getArticleDetail(widgetUrl.split("card/")[1])
-      .data?.toDomainModel(campaignRepository, campaignUrlNormalizer)
-      ?: throw IllegalStateException()
+  override suspend fun getArticle(articleId: String): Article = cachedUrls[articleId]
+    ?.let { it.split("card/")[1] }
+    ?.let { editorialRemoteDataSource.getArticleDetail(it) }
+    ?.data?.toDomainModel(campaignRepository, campaignUrlNormalizer)
+    ?: throw IllegalStateException()
```

Comment:
If the article is not cached, a network request could be performed to fetch that single article and cache it

Reply:
Not possible now, discussed on daily